coralproject/talk
# [CORL-1761] Show external moderation phases on the moderate card

## What does this PR do?

Adds markers, "moderate by" state, and "automated actions" details to the comment moderation cards in the mod queues.

![image](https://user-images.githubusercontent.com/5751504/122790278-64423800-d275-11eb-98b5-08b0ea339088.png)

## What changes to the GraphQL/Database Schema does this PR introduce?

- Adds external moderation phase metadata to the comment's metadata

## How do I test this PR?

- Create an external moderation phase in Admin > Config > Moderation
  - Will require making a server to call an endpoint. I suggest using a quick express app with a POST endpoint to return some status like: 
    ```
      app.post("/mod", (req, res) => {
        res.send({
          actions: [
            {
              actionType: "FLAG",
              reason: "COMMENT_DETECTED_TOXIC",
            },
            {
              actionType: "FLAG",
              reason: "COMMENT_DETECTED_SPAM",
            },
          ],
        });
      });
    ```
- Create some comments as a non-admin/mod user
- See the comments get flagged with whatever you put into your mod phase and that the comments are marked with tags of the mod phases in the moderate card
- You can also force them into specific queues by making the mod result return `status: "REJECTED"` or similar
